### ISSUE TITLE
Panel no longer has copy option

### ISSUE THREAD

**Issue by muddro1** [linuxmint/cinnamon#9052]
```
 * Cinnamon version (cinnamon --version): Cinnamon 4.4.5
   - Please specify if you are using the daily builds PPA (https://launchpad.net/~linuxmint-daily-build-tea
m/+archive/ubuntu/daily-builds).
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...)
 * Graphics hardware *and* driver used
 * 32 or 64 bit
 * Attach /home/<username>/.xsession-errors, or /var/log/syslog
 ```

**Issue**
Mint 19.3 Panel has removed the option to copy applets from one panel to another

**Steps to reproduce**

**Expected behaviour**
The ability to copy applet layout from one panel to another

**Other information**
worked in mint 19.2, disappeared in 19.3

**Comment by Odyseus** [comment]
Hello, @muddro1.

As reference, this option was removed in 943880fca68c0588e95a7040fabb34470054addd.

#### Commit message extract

> ...
> - Remove advanced niche options (copy/paste applet conf, clear applets)
> 
> About these niche options.. they were commented out in panel.js but the code was
> left and their callbacks are still in Cinnamon.
> 
> I'm not sure whether or not they're worth keeping. I suppose feedback will tell
> us that. If we're bringing them back, they shouldn't stay here, they should be
> implemented in a burger menu in cs_panel.py instead.

**Comment by muddro1** [comment]
Ahh, ok I see. Thanks for the response. Well, for what it's worth, +1 on bringing back that functionality somehow, whether in the burger or what not.

**Comment by Krafting** [comment]
I liked this option, I wish it was still present :(

**Comment by phillies** [comment]
Exactly, removing this functionality was a big mistake and makes cinnamon very uncomfortable in dual screen mode for me. Bring this function back!

**Comment by vbujanca** [comment]
Please bring back the applet copy feature as it's the only way to mirror or even have a proper/complete panel on a secondary monitor. As it is, cinnamon won't allow certain applets on secondary monitors or copy menu launchers on other panels.

Please and thank you!

**Comment by mtwebster** [comment]
Can you guys please detail what made this useful?  We removed it because presumably people aren't setting up panels every day.  If you are, I'd like to know why, and maybe there's an alternate solution for this.

Btw, the panel configuration follows the primary monitor.  For example, if you have a laptop, when unplugged obviously your laptop screen is primary, your primary monitor's panels are shown.  Let's say you then plug in to an external monitor, you can make that monitor your primary for *that* configuration, and the primary monitor panels will move there.  When you unplug that monitor, the configuration will switch back to the single monitor setup, and those panels will once more be on the laptop monitor.

I'm having trouble coming up with a scenario where this would be insufficient.  Yes, a couple of applets can't have multiple instances - for technical reasons, mainly.  I'm not saying there isn't one.  And I'd rather haver panel configurations be remembered like monitor configurations are, rather than this copy thing (which was broken).

**Comment by vbujanca** [comment]
It might be my setup is less common, such as i am using 2 external screens connected to the laptop. 
I am only using the external monitors and as a workaround I set one of the externals as primary and built a less endowed panel on the other. As these screens are quite large would like to be able to access some launchers/applets from both panels. I couldn't find a way to do that just yet, ie some applets are just disallowed, can't be dragged/dropped. Or I might've come across a cinnamon bug? (Using v. 4.4.8)

What I would really like to have is leaving the laptop display primary and just mirror the panels on each display.

**Comment by Krafting** [comment]
For me, the option was useful in the first setup of my computer, I have multiple display on a Desktop computer, it was faster for me to just configure a panel and copy the same configuration for the applets that support multiple instance to the other panel on my others monitors.

But I should admit this feature was really useful when it copied my "Grouped window list" settings to the other monitor, but now this applet doesn't show all windows on all monitor (the option is gone) so I need to copy the applets and create a second one so I can have all opened windows showing on all monitor
(the copy option was also useful for copying pinned apps)

Also I'd love to know why there is a restriction for some applets (the one that only allows one instance). I Understand that there are technical details that prevent multiple instance but I'm curious to have those details!

**Comment by severedsolo** [comment]
I realise this is a bit old but I'd like to throw a "+1" behind restoring the option for copying the panel configuration.

My use case is in a dual monitor setup. It's so much quicker to hit "copy configuration" and "paste" onto a new panel, rather than having to set the applets up manually. As a gamer I like to have the clock on both monitors so I can see what time it is at a glance, and it remembers my options (custom time format) and other configurations. (Both monitors are always connected, so the moving panel between primary etc was not an issue for me)

Like I said, this *can* be done manually, but it's a chore. Doing this on Ulyana involved it putting the applets in weird places, which then had to be moved manually, and the panel can be a bit finicky when doing drag and drop in panel edit mode.

As for the applets that can't be copied - in my experience they are things I'd count as "system tray" icons (like network manager etc) and not stuff that's critical to have on the other monitor anyway.

**Comment by CeuRoxo** [comment]
Apologies if it's bad form to post on such an old thread, but I just wanted to do the same as the last poster instead of creating a fresh thread for a feature request - I thought that would be poorer form. "+1" again for being able to have identical panels on two monitors.

My use case is much the same, but I'd go one step further and say I'd love to be able to have my system tray icons and notifications on both monitors at once, due to having different things in full-screen mode on each monitor at various times. It's a little inconvenient to have to alt+tab out of a game or come out of full screen on a video or stream to access those things when I have another monitor sitting here that could host those icons and allow me to see what's going on more easily. Personally, I'd also like there to be an option to have program icons on both panels, regardless of which monitor they are open on for similar reasons.

My current version is 
Cinnamon 5.2.7
Linux Mint 20.3

**Comment by iqzer0** [comment]
i loved this feature, was really easy setting up dual monitors :) hope it comes back :+1:

**Comment by Lurux** [comment]
This would be really useful for multi-monitor setups, since new monitors don't come with a default panel and there doesn't seem to be an option to "show panel on all monitors" atm

**Comment by JosephMcc** [comment]
Currently no plans to implement this. All feature requests now go here: https://github.com/orgs/linuxmint/discussions
